### PR TITLE
Config fixups

### PR DESCRIPTION
**Pull request by mudler**
While I was working on #532 I noticed that the config files currently in `cos-upgrade` could actually override what is passed explicitly by arguments.

This change also makes the whole more consistent and also allows to use only `/etc/os-release` as a single source of truth.

Relates to #224

### CODE REVIEW

File: packages/installer/upgrade.sh [hunk 1]
```diff
@@ -54,10 +54,23 @@ find_recovery() {
...
+    if [ -e "/etc/os-release" ]; then
+        source /etc/os-release
+    fi
+
     if [ -e "/etc/cos-upgrade-image" ]; then
         source /etc/cos-upgrade-image
     fi
 
+    if [ -e "/etc/cos/config" ]; then
+        source /etc/cos/config
```
Comment (davidcassany) [inline packages/installer/upgrade.sh:71]: Isn't it still overwriting what was passed in the command line? `find_upgrade_channel` happens right after evaluating the command line parameters. If the command line includes `--recovery` and `/etc/cos/config` includes `UPGRADE_RECOVERY=false` the recovery flag turns to be useless. I am unsure of what is expected to be in `/etc/cos/config` though

**Review by davidcassany — COMMENTED** [review]
Beyond the additional loaded files I am not sure I see a real behavior change.

File: packages/installer/upgrade.sh (hunk 1 above)
Comment (mudler) [reply packages/installer/upgrade.sh:71]: The image that is being passed from the CLI is being set right afterwards here: https://github.com/rancher-sandbox/cOS-toolkit/blob/f272f6e25e9121ca5a808ee45dd9f8982738bce8/packages/installer/upgrade.sh#L65

UPGRADE_RECOVERY, even if could be set here it shouldn't be part of the config, it's just to drive the CLI selection

Comment (davidcassany) [reply packages/installer/upgrade.sh:71]: OH I see, this is for the image parameter and the config associated with the upgrade image. All clear now, makes complete sense.

**Review by davidcassany — APPROVED** [review]
LGTM

**Comment by kkaempf** [comment]
There should be a documentation PR explaining these files and changes.

**Comment by mudler** [comment]
> There should be a documentation PR explaining these files and changes.

post-mortems?! https://github.com/rancher-sandbox/cos-toolkit-docs/pull/24 :)

**Pull request merged**